mc-sema/runtime/linux_amd64_callback.c

```diff
@@ -37,22 +37,27 @@ uint64_t __mcsema_get_max_stack() {
     uint64_t start;
     uint64_t end;
     while( fgets(line, sizeof(line), p) != NULL ) {
-        if(NULL != strstr(line, "[stack]")) {
-            if(sscanf(line, "%lx-%lx", &start, &end) == 2) {
+        if(sscanf(line, "%lx-%lx", &start, &end) == 2) {
+            if(start <= val  && val <= end)  {
                 fclose(p);
                 return end;
-            } else {
-                fclose(p);
-                return 0;
             }
+        } else {
+            break;
         }
     }
 
     fclose(p);
-
     return 0;
```

Comment:
There is a check for this returning 0 later down the line; but I can see how this should probably assert or at least give some notification